Assets/Tests/EditMode/LightSwitchTests.cs

```diff
@@ -34,61 +36,75 @@ public void ResetEnumeratorToZero()
+        Assert.IsTrue(lightsSwitch.LightStatus(0, 1));
+        Assert.IsFalse(lightsSwitch.LightStatus(2, 1));
+        Assert.IsTrue(lightsSwitch.LightStatus(4, 1));
     }
 
     [Test]
-    public void ThrowExceptionForInvalidBlackoutInterval()
+    public void LightsSwitch_ResetBehavior()
     {
-        Assert.Throws<ArgumentException>(() => new LightsSwitch(5, new List<int> { -1, 6 }));
+        var lightsSwitch = new LightsSwitch(10, new List<int> { 1, 3 });
+        lightsSwitch.LightStatus(1, 1);
+        lightsSwitch.Reset();
+        Assert.IsTrue(lightsSwitch.LightStatus(0, 1));
     }
 
     [Test]
-    public void ReturnTrueIfNoBlackouts()
+    public void LightsSwitch_HandleEmptyBlackoutList()
     {
         var lightsSwitch = new LightsSwitch(10, new List<int>());
-
-        bool lightStatus = lightsSwitch.LightStatus(0, 1);
-
-        Assert.IsTrue(lightStatus);
+        Assert.IsTrue(lightsSwitch.LightStatus(0, 1));
+        Assert.IsTrue(lightsSwitch.LightStatus(5, 1));
     }
 
     [Test]
-    public void HandleMultipleBlackoutsProperly()
+    public void LightsSwitch_HandleDifferentAgentDecisionIntervals()
     {
-        var blackouts = new List<int> { 1, 2, 3 };
-        var lightsSwitch = new LightsSwitch(10, blackouts);
-
-        Assert.IsTrue(lightsSwitch.LightStatus(0, 1));
-        Assert.IsFalse(lightsSwitch.LightStatus(1, 1));
-        Assert.IsTrue(lightsSwitch.LightStatus(2, 1));
-        Assert.IsFalse(lightsSwitch.LightStatus(3, 1));
+        var lightsSwitch = new LightsSwitch(20, new List<int> { 2, 4 });
+        Assert.IsTrue(lightsSwitch.LightStatus(0, 2));
+        Assert.IsFalse(lightsSwitch.LightStatus(4, 2));
+        Assert.IsTrue(lightsSwitch.LightStatus(8, 2));
     }
 
     [Test]
-    public void ThrowExceptionForInvalidStepOrAgentDecisionInterval()
+    public void LightsSwitch_ThrowExceptionForInvalidBlackoutSequence()
     {
-        var lightsSwitch = new LightsSwitch(10, new List<int> { 1 });
+        Assert.Throws<ArgumentException>(() => new LightsSwitch(20, new List<int> { 10, 1 }));
+    }
 
-        Assert.Throws<ArgumentException>(() => lightsSwitch.LightStatus(-1, 1));
-        Assert.Throws<ArgumentException>(() => lightsSwitch.LightStatus(1, 0));
+    [Test]
+    public void LightsSwitch_HandleInfiniteBlackoutsWithNegativeNumber()
+    {
+        var lightsSwitch = new LightsSwitch(100, new List<int> { -20 });
+        Assert.IsTrue(lightsSwitch.LightStatus(0, 1));
+        Assert.IsTrue(lightsSwitch.LightStatus(19, 1));
+        Assert.IsFalse(lightsSwitch.LightStatus(20, 1));
+        Assert.IsFalse(lightsSwitch.LightStatus(39, 1));
+        Assert.IsTrue(lightsSwitch.LightStatus(40, 1));
+        Assert.IsTrue(lightsSwitch.LightStatus(59, 1));
+        Assert.IsFalse(lightsSwitch.LightStatus(60, 1));
+        Assert.IsFalse(lightsSwitch.LightStatus(79, 1));
+        Assert.IsTrue(lightsSwitch.LightStatus(80, 1));
```

Comment:
I think the best approach here is to avoid using loops and provide a more clearer SUT. I will refactor this into something more readable. Basically the numbers represent specific time steps where the light status changes based on the blackout interval. In this test, the blackout interval is 20 (since -20 indicates an infinite blackout every 20 steps). The light status should toggle every 20 steps.

Sorry perhaps I should have chosen a better and more informative test name.